### PR TITLE
Add overload for Connect() to allow other libvirt drivers

### PR DESCRIPTION
**Pull request by baez90**
Issue #116

### CODE REVIEW

File: .gitignore
Comment (benlemasurier) [inline .gitignore:1]: did you mean to include this?

File: libvirt.go
Comment (benlemasurier) [inline libvirt.go:44]: I'm not sure we gain much with an additional type here. Not a blocker for me, happy for others to weigh in.

File: libvirt_test.go
Comment (benlemasurier) [inline libvirt_test.go:38]: :heart:

File: .gitignore
Comment (baez90) [reply .gitignore:1]: no I wanted to exclude it to make sure I don't mess the repo up with some GoLand specific stuff

File: libvirt.go
Comment (baez90) [reply libvirt.go:44]: Thought it makes stuff easier for beginners to find the included constants with the IDE and type based predictions and because it's a string anyway it does not block anyone to provide custom URIs but I could of course change this?

File: libvirt.go
Comment (baez90) [reply libvirt.go]: done

File: libvirt.go
Comment (baez90) [reply libvirt.go]: done

File: .editorconfig
Comment (benlemasurier) [inline .editorconfig:1]: no need to include this file

File: libvirt.go
Comment (benlemasurier) [reply libvirt.go:45]: good enough for me :+1:

File: .editorconfig
Comment (baez90) [reply .editorconfig]: removed it again - just thought it would be nice to enforce the line length limit :blush:

**Comment by benlemasurier** [comment]
lgtm - once travis[1] is happy I'll get this merged. Thanks, very much appreciated!

[1] - super slow at the moment. I'm in the process of switching over to GH Actions as we speak

File: libvirt.go
Comment (trapgate) [reply libvirt.go:45]: I'm OK with using a type here, but the name stutters: it's `libvirt.LibvirtURI`. I suggest shortening the type name to just `URI`.

File: libvirt.go
Comment (trapgate) [inline libvirt.go:45]: I'm OK with using a type here, but the name stutters: it's libvirt.LibvirtURI. I suggest shortening the type name to just URI.

**Comment by trapgate** [comment]
:+1: Thanks for the PR, I think this is a nice improvement.

File: libvirt.go
Comment (baez90) [reply libvirt.go:45]: what about `DriverURI` to avoid confusion?

Comment (benlemasurier) [reply libvirt.go:45]: `ConnectURI`? naming is the worst.

Comment (baez90) [reply libvirt.go]: :joy: true that - I'll change it to `ConnectURI`

**Comment by baez90** [comment]
Tests should be fixed now, I just haven't noticed that there's not `libvirt.sock` in the test environment :sweat_smile: so I changed the test to use the TCP socket (had to update my local setup though which was the reason why I wanted to use the socket in the first place I think...) but I hope this can be merged as soon as the builds are finished.

Thanks for your tremendous fast feedback and input!

**Review by benlemasurier — APPROVED** [review]

**Comment by MattIPv4** [comment]
Hey, @baez90 - Thanks a ton for this sweet PR! 😄

Would you please shoot me an email when you get a chance?
mcowley at digitalocean dot com 🎉

**Pull request merged**
